poly/joint.go

```diff
@@ -0,0 +1,284 @@
+package poly
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"io"
+)
+
+// This package provides  a dealer-less distributed verifiable secret sharing
+// using Pedersen VSS scheme as explained in "Provably Secure Distributed
+// Schnorr Signatures and a (t, n) Threshold Scheme for Implicit Certificates"
+// This file is only responsible for the setup of a shared secret among n peers.
+// The output is a global public polynomial (pubPoly) and a secret share for
+// each peers.
+
+// PolyInfo describe the information needed to construct (and verify) a matrixShare
+type PolyInfo struct {
+	// How many peer do we need to reconstruct a secret
+	T int
+	// How many peers do we need to verify
+	R int
+	// How many peers are collaborating into constructing the shared secret ( i.e. MatrixShare is of size NxN)
+	N int
+}
+
+// Represent the output of a VSS Pedersen scheme : a global public polynomial and a share of its related priv poly
+// for a peer
+type SharedSecret struct {
+
+	// The shared public polynomial
+	Pub *PubPoly
+
+	// The share of the shared secret
+	Share *abstract.Secret
+
+	// The index of this share regarding the secret private poly / pub poly
+	// i.e. it is the same as the receiver's index
+	Index int
+}
+
+// Dealer is a peer that will create a promise and distribute it to each receivers needed
+type Dealer struct {
```

Comment:
Similarly, please clarify the comment a bit further, and should these fields be public or private?